subcommand/server-acl-init/command.go

```diff
@@ -391,27 +391,27 @@ func (c *Command) Run(args []string) int {
 	}
 
 	if c.flagCreateSyncToken {
-		syncRules, err := c.syncRules()
+		syncRules, localToken, err := c.syncRules()
 		if err != nil {
 			c.Log.Error("Error templating sync rules", "err", err)
 			return 1
 		}
 
-		err = c.createLocalACL("catalog-sync", syncRules, consulDC, consulClient)
+		err = c.createACL("catalog-sync", syncRules, localToken, consulDC, consulClient)
 		if err != nil {
 			c.Log.Error(err.Error())
 			return 1
 		}
 	}
 
 	if c.flagCreateInjectToken {
-		injectRules, err := c.injectRules()
+		injectRules, localToken, err := c.injectRules()
 		if err != nil {
 			c.Log.Error("Error templating inject rules", "err", err)
 			return 1
 		}
 
-		err = c.createLocalACL("connect-inject", injectRules, consulDC, consulClient)
+		err = c.createACL("connect-inject", injectRules, localToken, consulDC, consulClient)
```

Comment:
Why do we need to return the boolean signaling whether the policy/token should be local above? Couldn't we just pass `!c.flagEnableNamespaces`?

Reply:
Yes but I did it this way because it's not the fact that we're using namespaces that means we need a global token (see the client policy). It's the fact that this specific component needs to be able to create namespaces. I thought it best to put the "knows that this component needs to create namespaces" knowledge into the rules function because it's already giving the rules operator:write because it knows this component needs to create namespaces so the knowledge is contained within that function.

Reply:
Interesting. I found it a bit confusing because there's nothing that function does to actually make that decision; it's especially evident in the tests where we're testing for boolean values but they are always the opposite of enableNamespaces. You could separate it into a variable to make it more clear.

If you feel strongly about keeping in the function, I would recommend renaming the variable to something more boolean-like, e.g. `isLocal`, because on an initial read I thought it's an actual token.

Reply:
That's fair. I've moved it out of rules and into the `if` statement.